src/Features/CSharp/Portable/Completion/CompletionProviders/PropertySubPatternCompletionProvider.cs

```diff
@@ -0,0 +1,177 @@
+
+            if (token.Kind() != SyntaxKind.CommaToken && token.Kind() != SyntaxKind.OpenBraceToken)
+            {
+                return default;
+            }
+
+            if (token.Parent?.IsKind(SyntaxKind.PropertySubpattern) == false ||
+                token.Parent.Parent?.IsKind(SyntaxKind.PropertyPattern) == false)
+            {
+                return default;
+            }
+
+            // is Goo { $$
+            // is Goo { P1: 0, $$
+            var propertyPattern = (PropertyPatternSyntax)token.Parent.Parent;
+            var patternType = propertyPattern.Type;
+            if (patternType != null)
+            {
+                var typeSymbol = (ITypeSymbol)semanticModel.GetSymbolInfo(patternType, cancellationToken).Symbol;
+                return (typeSymbol, token.GetLocation());
+            }
+
+            // e is { $$
+            // e is { P1: 0, $$
+            if (propertyPattern.IsParentKind(SyntaxKind.IsPatternExpression))
+            {
+                var isPattern = (IsPatternExpressionSyntax)propertyPattern.Parent;
+                var typeSymbol = semanticModel.GetTypeInfo(isPattern.Expression, cancellationToken).Type;
+                if (typeSymbol != null)
+                {
+                    return (typeSymbol, token.GetLocation());
+                }
+            }
+
+            // switch (e) { case { $$
+            // switch (e) { case { P1: 0, $$
+            if (propertyPattern.IsParentKind(SyntaxKind.CasePatternSwitchLabel))
+            {
+                var casePattern = (CasePatternSwitchLabelSyntax)propertyPattern.Parent;
+                var switchStatment = casePattern.Parent?.Parent as SwitchStatementSyntax;
+                if (switchStatment != null && switchStatment.Expression != null)
+                {
+                    var typeSymbol = semanticModel.GetTypeInfo(switchStatment.Expression).Type;
+                    if (typeSymbol != null)
+                    {
+                        return (typeSymbol, token.GetLocation());
+                    }
+                }
+            }
+
+            // e is { P1: { $$
+            // e is { P1: { P2: 2, $$
+            if (propertyPattern.IsParentKind(SyntaxKind.SubpatternElement))
+            {
+                var containingProperty = ((SubpatternElementSyntax)propertyPattern.Parent).NameColon?.Name;
+                if (containingProperty != null)
+                {
+                    var symbol = semanticModel.GetSymbolInfo(containingProperty).Symbol;
+
+                    switch (symbol)
+                    {
+                        case null:
+                            break;
+                        case IPropertySymbol propertySymbol:
+                            return (propertySymbol.Type, token.GetLocation());
+                        case IFieldSymbol fieldSymbol:
+                            return (fieldSymbol.Type, token.GetLocation());
+                    }
+                }
```

Comment:
what about this form: `e is { P1: SomeOtherType { P2: 2, $$` where we don't care about the type of `P1`. is this handled above? is this tested?